.github/workflows/distcheck.yml

```diff
@@ -0,0 +1,123 @@
+# Copyright (C) 2000 - 2022 Daniel Stenberg, <daniel@haxx.se>, et al.
+#
+# SPDX-License-Identifier: curl
+
+name: dist
+
+on:
+  push:
+    branches:
+    - master
+    - '*/ci'
+  pull_request:
+    branches:
+    - master
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  maketgz:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+    - uses: actions/checkout@v3
+
+    - run: df -h
+
+    - run: autoreconf -fi
+      name: 'autoreconf'
+
+    - run: df -h
+
+    - run: ./configure --without-ssl
+      name: 'configure'
+
+    - run: df -h
+
+    - run: make V=1 && make V=1 clean
+      name: 'make and clean'
+
+    - run: df -h
+
+    - run: /bin/sh -x ./maketgz 99.98.97
+      name: 'maketgz'
+
+    - run: df -h
+
+    - uses: actions/upload-artifact@v2
+      with:
+        name: 'release-tgz'
+        path: 'curl-99.98.97.tar.gz'
+
+  verify-in-tree:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    needs: maketgz
+    steps:
+    - uses: actions/download-artifact@v2
+      with:
+        name: 'release-tgz'
+
+    - run: df -h
+
+    - run: |
+        echo "::stop-commands::$(uuidgen)"
+        tar xvf curl-99.98.97.tar.gz
+        pushd curl-99.98.97
+        ./configure --prefix=$HOME/temp --without-ssl
+        make
+        make TFLAGS=1 test
+        make install
+        popd
+        # basic check of the installed files
+        bash scripts/installcheck.sh $HOME/temp
```

Comment:
This should run the script from the git tree since the `installcheck.sh` itself is not included in the tarball.

Maybe just:

`bash ../scripts/installcheck.sh $HOME/temp`